File: src/Content/NetCoreTool.Template.WebApi/CSharp/Services/RabbitListenerService.cs
#if (MessagingRabbitMqListener)
using Microsoft.Extensions.Logging;
using Steeltoe.Messaging.RabbitMQ.Attributes;

namespace Company.WebApplication.CS.Services
{
    public class RabbitListenerService
    {
        private ILogger _logger;

        public RabbitListenerService(ILogger<RabbitListenerService> logger)
        {
            _logger = logger;
        }

        [RabbitListener("steeltoe_message_queue")]
        public void ListenForAMessage(string msg)
        {
            _logger.LogInformation($"Received the message '{msg}' from the queue.");
        }
    }
}
#endif

File: src/Content/NetCoreTool.Template.WebApi/FSharp/Services/RabbitListenerService.fs
namespace Company.WebApplication.FS.Services
#if (MessagingRabbitMqListener)

open Microsoft.Extensions.Logging
open Steeltoe.Messaging.RabbitMQ.Attributes

type RabbitListenerService (logger : ILogger<RabbitListenerService>) =
    let logger = logger

    [<RabbitListener("steeltoe_message_queue")>]
    member _.ListenForAMessage(message : string) =
        logger.LogInformation(message)
#endif
